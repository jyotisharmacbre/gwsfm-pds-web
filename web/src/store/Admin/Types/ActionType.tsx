export enum ActionType {
	GET_PROJECT_PARAMETERS_SUCCESS = 'GET_PROJECT_PARAMETERS_SUCCESS',
	GET_PROJECT_PARAMETERS_ERROR = 'GET_PROJECT_PARAMETERS_ERROR'
}
