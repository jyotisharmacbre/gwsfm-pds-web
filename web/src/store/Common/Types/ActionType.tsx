export enum ActionType {
  DEFAULT = 'DEFAULT'
}
