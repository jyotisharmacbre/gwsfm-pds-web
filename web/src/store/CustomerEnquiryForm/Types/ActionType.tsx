export enum ActionType {
  DEFAULT = 'DEFAULT',
  PROJECT_ADD = 'PROJECT_ADD',
  PROJECT_ADD_ERROR = 'PROJECT_ADD_ERROR',
  PROJECT_EDIT_SUCCESS = 'PROJECT_EDIT_SUCCESS',
  GET_ENQUIRY_OVERVIEW_SUCCESS = 'GET_ENQUIRY_OVERVIEW_SUCCESS',
  GET_ENQUIRY_OVERVIEW_ERROR = 'GET_ENQUIRY_OVERVIEW_ERROR',
  GET_PROJECT_DETAIL_SUCCESS = 'GET_PROJECT_DETAIL_SUCCESS',
  GET_PROJECT_DETAIL_ERROR = 'GET_PROJECT_DETAIL_ERROR',
  RESET_PROJECT_DETAIL_STATE = 'RESET_PROJECT_DETAIL_STATE',
  SET_PROJECT_ID_STATE = 'SET_PROJECT_ID_STATE'
}
