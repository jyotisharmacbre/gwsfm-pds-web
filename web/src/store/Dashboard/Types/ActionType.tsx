export enum ActionType {
  PROJECT_DASHBOARD_GRID_ERROR = 'PROJECT_DASHBOARD_GRID_ERROR',
  PROJECT_DASHBOARD_GRID_DETAILS = 'PROJECT_DASHBOARD_GRID_DETAILS'
}
