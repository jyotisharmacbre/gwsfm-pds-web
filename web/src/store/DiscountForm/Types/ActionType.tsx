export enum ActionType {
	DISCOUNT_FORM_DATA_ADD = 'DISCOUNT_FORM_DATA_ADD',
	DISCOUNT_FORM_DATA_ERROR = 'DISCOUNT_FORM_DATA_ERROR',
	DISCOUNT_FORM_DATA_EDIT = 'DISCOUNT_FORM_DATA_EDIT',
	RESET_DISCOUNT_FORM_STATE = 'RESET_DISCOUNT_FORM_STATE',
	RESET_DISCOUNT_FORM_STATE_NOTIFIER = 'RESET_DISCOUNT_FORM_STATE_NOTIFIER',
	DISCOUNT_FORM_DATA_GET = 'DISCOUNT_FORM_DATA_GET',
	DISCOUNT_FORM_DATA_GET_ERROR = 'DISCOUNT_FORM_DATA_GET_ERROR'
}
