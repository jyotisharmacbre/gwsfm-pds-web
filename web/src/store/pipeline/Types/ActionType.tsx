export enum ActionType {
    PROJECT_PIPELINE_GRID_ERROR = 'PROJECT_PIPELINE_GRID_ERROR',
    PROJECT_PIPELINE_GRID_DETAILS = 'PROJECT_PIPELINE_GRID_DETAILS',
    GET_PROJECT_CHART_SUMMARY_SUCCESS = 'GET_PROJECT_CHART_SUMMARY_SUCCESS',
    GET_PROJECT_CHART_SUMMARY_ERROR = 'GET_PROJECT_CHART_SUMMARY_ERROR',
    SET_LOADING_TRUE = "SET_LOADING_TRUE"
}
