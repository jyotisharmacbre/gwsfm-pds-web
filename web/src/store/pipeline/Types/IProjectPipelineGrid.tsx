import moment from 'moment';

export interface IProjectPipelineGrid {
  projectId: string;
  name: string;
  projectOwner: string;
  contractorId: number;
  probabilityOfWinning: number;
  lastModified: string;
  status: string;
  commenceDate: string;
  approxValue: 0;
  contractTypeId: 0;
  cdmNotifiable: false;
  soldMargin: string;
  weightedTCV: string;
  rank: string;
}
