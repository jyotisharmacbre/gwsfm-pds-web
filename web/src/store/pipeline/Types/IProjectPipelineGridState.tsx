import { IProjectPipelineGrid } from './IProjectPipelineGrid';
import { Notify } from '../../../helpers/constants';

export interface IProjectPipelineGridState {
  pipelineDetails: Array<IProjectPipelineGrid>;
  error: string | null;
}
