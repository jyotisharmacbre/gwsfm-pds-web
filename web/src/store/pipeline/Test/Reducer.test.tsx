import { ActionType } from './../Types/ActionType';
import projectPipelineDetailReducer, { initialState } from '../Reducer';

initialState.data[0].lastModified = '';

describe('Pipeline reducer test cases', () => {

    it('should handle PROJECT_PIPELINE_GRID_DETAILS ', () => {
        const PROJECT_PIPELINE_GRID_DETAILS: any = {
            type: ActionType.PROJECT_PIPELINE_GRID_DETAILS,
            payload: initialState
        };
        expect(projectPipelineDetailReducer(initialState, PROJECT_PIPELINE_GRID_DETAILS)).toMatchSnapshot();
    });

    it('should handle PROJECT_PIPELINE_GRID_ERROR ', () => {
        const PROJECT_PIPELINE_GRID_ERROR: any = {
            type: ActionType.PROJECT_PIPELINE_GRID_ERROR,
            payload: { error: true }
        };
        expect(projectPipelineDetailReducer(initialState, PROJECT_PIPELINE_GRID_ERROR)).toMatchSnapshot();
    });
})