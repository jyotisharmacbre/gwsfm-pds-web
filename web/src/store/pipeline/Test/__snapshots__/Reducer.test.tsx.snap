// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pipeline reducer test cases should handle PROJECT_PIPELINE_GRID_DETAILS  1`] = `
Object {
  "data": Array [
    Object {
      "approxValue": 0,
      "cdmNotifiable": false,
      "commenceDate": "",
      "contractTypeDescription": "",
      "contractTypeId": 0,
      "contractorId": "-1",
      "headOfProject": "",
      "lastModified": "",
      "name": "",
      "probabilityOfWinning": 1,
      "projectId": "",
      "projectOwner": "",
      "projectRefId": 0,
      "projectStatusDescription": "",
      "soldMargin": "",
      "status": "",
      "weightedTCV": "",
    },
  ],
  "error": null,
  "loading": false,
  "pipelineDetails": Object {
    "data": Array [
      Object {
        "approxValue": 0,
        "cdmNotifiable": false,
        "commenceDate": "",
        "contractTypeDescription": "",
        "contractTypeId": 0,
        "contractorId": "-1",
        "headOfProject": "",
        "lastModified": "",
        "name": "",
        "probabilityOfWinning": 1,
        "projectId": "",
        "projectOwner": "",
        "projectRefId": 0,
        "projectStatusDescription": "",
        "soldMargin": "",
        "status": "",
        "weightedTCV": "",
      },
    ],
    "error": null,
    "loading": false,
    "projectChartSummary": Object {
      "data": Array [],
      "error": null,
      "loading": false,
    },
    "totalNumberOfRecord": 1,
  },
  "projectChartSummary": Object {
    "data": Array [],
    "error": null,
    "loading": false,
  },
  "totalNumberOfRecord": 1,
}
`;

exports[`Pipeline reducer test cases should handle PROJECT_PIPELINE_GRID_ERROR  1`] = `
Object {
  "data": Array [
    Object {
      "approxValue": 0,
      "cdmNotifiable": false,
      "commenceDate": "",
      "contractTypeDescription": "",
      "contractTypeId": 0,
      "contractorId": "-1",
      "headOfProject": "",
      "lastModified": "",
      "name": "",
      "probabilityOfWinning": 1,
      "projectId": "",
      "projectOwner": "",
      "projectRefId": 0,
      "projectStatusDescription": "",
      "soldMargin": "",
      "status": "",
      "weightedTCV": "",
    },
  ],
  "error": undefined,
  "loading": false,
  "notify": 2,
  "pipelineDetails": Array [],
  "projectChartSummary": Object {
    "data": Array [],
    "error": null,
    "loading": false,
  },
  "totalNumberOfRecord": 0,
}
`;
